### PR TITLE
Improve navigation and clarify content.

### PR DESCRIPTION
**Pull request by brainwane**
- De-duplicate content that lives elsewhere and point to the current canonical source. (We're updating README.prod.md much more often than server.html.)
- Add link to zulip.com/api in top navbar.
- Add more guidance for new contributors.
- Fix several obsolete or unclear links.
- Fix formatting, whitespace, punctuation, and style.

Fixes: zulip/zulip#668.

### CODE REVIEW

File: server.html [hunk 1]
```diff
@@ -85,92 +92,10 @@
...
-            <p>Download
-            the <a href="https://www.zulip.com/dist/releases/zulip-server-1.3.12.tar.gz">binary
-            tarball</a>, Verify its sha1sum matches, extract it, and
-            move the resulting directory to /root/zulip:</p>
-            <div class="terminal">
-              cd /root<br />
-              wget https://www.zulip.com/dist/releases/zulip-server-1.3.12.tar.gz<br />
-              echo "95f65a74f7447b5b4c46e7ed487fca50ff7a2c72 zulip-server-1.3.12.tar.gz" | sha1sum -c<br />
-              tar -xf zulip-server-1.3.12.tar.gz<br />
-              mv zulip-server-1.3.12 zulip<br />
```
Comment (timabbott) [inline server.html]: Currently, this place is the only place where we document the sha1sums of the Zulip tarballs.  We need to put that somewhere before we can get rid of this...

File: contribute.html [hunk 2]
```diff
@@ -47,40 +48,75 @@
...
+          more. Here's just a small sampling of ways you can get
+          started:</p>
           <ul>
-            <li>Write an integration for your favorite service</li>
-            <li>Write a docker file to make Zulip available to dockerized servers</li>
+            <li><a href="https://zulip.readthedocs.io/en/latest/integration-guide.html">Write
+            an integration for your favorite service</a></li>
+            <li>Improve
+            our <a href="https://github.com/zulip/zulip/blob/master/Dockerfile">Dockerfile</a>
+            to make Zulip available to dockerized servers</li>
```
Comment (timabbott) [inline contribute.html:70]: Would link to translations here, perhaps?

File: contribute.html [hunk 3]
```diff
@@ -47,40 +48,75 @@
...
           </ul>
+	  <p>We maintain several more repositories for integrations
+	  and glue
+	  code; <a href="https://github.com/zulip/zulip/blob/master/README.md">see
+	  our README</a>.</p>
         </section>
+	<section>
+	  <h3>Security</h3>
+	  <p>Please report any security issues you discover to
+	  zulip-security@googlegroups.com.</p>
```
Comment (timabbott) [inline contribute.html:90]: s/security issues/potential security issues/

**Comment by timabbott** [comment]
I merged the 4 commits in this branch that I didn't have any comments on.  Let me know when you have the rest fixed up @brainwane !

File: contribute.html (hunk 2 above)
Comment (brainwane) [inline contribute.html:70]: Addressed -- thanks.

File: server.html (hunk 1 above)
Comment (brainwane) [inline server.html]: Understood. I kept the sha1sums and got rid of the rest of the duplicated content.

**Comment by brainwane** [comment]
Thanks @timabbott -- the updated branch is better thanks to your comments, and ready to merge.

File: contribute.html [hunk 4]
```diff
@@ -56,11 +56,21 @@
...
-            <li>Write a docker file to make Zulip available to dockerized servers</li>
+            <li><a href="https://zulip.readthedocs.io/en/latest/integration-guide.html">Write
+            an integration for your favorite service</a></li>
+            <li>Improve
+            our <a href="https://github.com/zulip/zulip/blob/master/Dockerfile">Dockerfile</a>
+            to make Zulip available to dockerized servers</li>
             <li>Add a service monitoring dashboard</li>
+	    <li><a href="https://zulip.readthedocs.io/en/latest/translating.html">Translate</a>
+	    user-visible words into another language</li>
             <li>Add your favorite emoji</li>
```
Comment (timabbott) [inline contribute.html]: Let's get rid of emoji and dashboard and replace them with things like "contribute to the roadmap"

Or, thinking about this more, do you think we should just dump this page and make it a link to (potentially an improved version of) http://zulip.readthedocs.io/en/latest/readme-symlink.html#ways-to-contribute ?

**Comment by timabbott** [comment]
On server.html, what do you think about putting the "download" section above the "Installation instructions" section?  Seems like that might be cleaner.

File: contribute.html (hunk 4 above)
Comment (brainwane) [inline contribute.html]: I'll slim this page way down and link to http://zulip.readthedocs.io/en/latest/readme-symlink.html#ways-to-contribute -- good idea -- but I still think it's a good idea to have this as a standalone zulip.org page, one that's got our branding. And I like that we have an opportunity here to display the zulip-security@googlegroups.com address **on** zulip.org, where it's easy to find for someone who wants/needs it.

**Comment by brainwane** [comment]
@timabbott I tried it that way, and yup, you're right. Will do.

**Comment by brainwane** [comment]
@timabbott Ready for you to review again.

**Comment by timabbott** [comment]
Merged the server.html stuff!  Will read through the rest (and get Luke to deploy) later.  

One small note on language; I think throughout the project we've been using "development" rather than abbreviating to "dev", and I guess we should be consistent about that (though the place where you add use of "dev" is deleted in the next commit, so whatever).

**Comment by brainwane** [comment]
@timabbott Aha, yes, I realize now that I didn't adhere to this convention; I'll be more careful in the future. Thanks.

**Comment by timabbott** [comment]
Merged this.  I think we'll want to do more work on improving that contribute page; I don't think the list of project ideas is as good as it could be.  But it's definitely better than before.

**Pull request closed without merging**